src/devices/Common/ClassVisibility.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+// These instructions make the classes public for building the individual projects.
+// The default visibility is "internal" so they stay hidden if this section is removed for the final library build.
+#if !BUILDING_IOT_DEVICE_BINDINGS
+
+#pragma warning disable SA1403 // File may only contain a single namespace
+
+public partial class Interop
```

Comment:
Honestly I don't love the fact of separating access modifiers from their implementation using partials. Given you would only have one instace per file, why not instead just add the ifdef on the class implementation itself to select if the class is either public or itnernal? I think having it separate like this will be harder to maintain and prone to errors.

Reply:
FWIW, we are already using that approach with at least three helper classes in common, I think we should use the same with the rest.

Reply:
This was @krwq's suggestion, and I think it's nice. It reduces the boilerplate on many files, and I think it reduces the possibility of errors. One would clearly not by accident change this file, but when one changes one of the other files (and i.e. adds an access specifier there) a build error would result. We could make this a bit more waterproof by anyway adding the explicit internal case.